02-express-tutorial/answers/final/007-params-query.js

```diff
@@ -0,0 +1,84 @@
+const express = require('express');
+const app = express();
+const {products} = require('./data');
+
+//1. Basic express query to display Products Json data on home page
+// app.get('/',(req,res)=>{
+//     res.json(products)
+// });
+
+//2 Basic Express query to display Home Page and a Link to Products,which will take us to '/api/products 
+app.get('/', (req, res) => {
+    res.send('<h1> Home Page</h1><a href="/api/products">products</a>')
+  })
+
+//3 Express query to display SOME JSON data for all products, and not all of the data for that product (i.e, not the 'description')
+app.get('/api/products',(req, res)=>{
+    const newProducts = products.map((p)=>{
+        const{id, name, image} = p;
+        return{id, name, image}
+    })
```

Comment:
Best practice is to avoid single character parameter names and be very explicit. `producto` > `p`, and your future self will thank you for making it very obvious 6 months later when you have to look back at code you wrote. :)

Reply:
Corrected.